pkg/app/handlers/config.go

```diff
@@ -0,0 +1,114 @@
+	DirectoryModelServiceURL    string `json:"directoryModelServiceUrl,omitempty"`
+}
+
+type TopazCfgV1 struct {
+	AsertoDirectoryURL       string `json:"asertoDirectoryUrl"`
+	AuthorizerServiceURL     string `json:"authorizerServiceUrl"`
+	AuthorizerAPIKey         string `json:"authorizerApiKey"`
+	DirectoryAPIKey          string `json:"directoryApiKey"`
+	DirectoryTenantID        string `json:"directoryTenantId"`
+	AsertoDirectoryReaderURL string `json:"asertoDirectoryReaderUrl,omitempty"`
+	AsertoDirectoryWriterURL string `json:"asertoDirectoryWriterUrl,omitempty"`
+	AsertoDirectoryModelURL  string `json:"asertoDirectoryModelUrl,omitempty"`
+}
+
+type TopazCfgV2 struct {
+	Type    string `json:"configType"`
+	Name    string `json:"name"`
+	Address string `json:"address"`
+	*TopazCfg
+}
+
+type CfgV2Response struct {
+	ReadOnly bool          `json:"readOnly"`
+	Configs  []*TopazCfgV2 `json:"configs"`
+}
+
+func ConfigHandler(confServices *TopazCfg) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		v1Cfg := &TopazCfgV1{
+			AsertoDirectoryURL:       confServices.DirectoryServiceURL,
+			AuthorizerServiceURL:     confServices.AuthorizerServiceURL,
+			AuthorizerAPIKey:         confServices.AuthorizerAPIKey,
+			DirectoryAPIKey:          confServices.DirectoryAPIKey,
+			DirectoryTenantID:        confServices.DirectoryTenantID,
+			AsertoDirectoryReaderURL: confServices.DirectoryReaderServiceURL,
+			AsertoDirectoryWriterURL: confServices.DirectoryWriterServiceURL,
+			AsertoDirectoryModelURL:  confServices.DirectoryModelServiceURL,
+		}
+
+		buf, _ := json.Marshal(v1Cfg)
+		writeJSON(buf, w, r)
+	}
+}
+
+func ConfigHandlerV2(confServices *TopazCfg) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		authorizerAPIKey := ""
+		directoryAPIKey := ""
+		authenticatedUser := r.Context().Value(AuthenticatedUser)
+		if authenticatedUser != nil && authenticatedUser.(bool) {
+			authorizerAPIKey = confServices.AuthorizerAPIKey
+			directoryAPIKey = confServices.DirectoryAPIKey
+		}
+
+		cfg := &TopazCfg{
+			AuthorizerServiceURL:        confServices.AuthorizerServiceURL,
+			AuthorizerAPIKey:            authorizerAPIKey,
+			DirectoryServiceURL:         confServices.DirectoryServiceURL,
+			DirectoryAPIKey:             directoryAPIKey,
+			DirectoryTenantID:           confServices.DirectoryTenantID,
+			DirectoryReaderServiceURL:   confServices.DirectoryReaderServiceURL,
+			DirectoryWriterServiceURL:   confServices.DirectoryWriterServiceURL,
+			DirectoryImporterServiceURL: confServices.DirectoryImporterServiceURL,
+			DirectoryExporterServiceURL: confServices.DirectoryExporterServiceURL,
+			DirectoryModelServiceURL:    confServices.DirectoryModelServiceURL,
+		}
+
+		cfgV2 := &TopazCfgV2{
+			Type:     "auto",
+			Name:     "Topaz Config",
+			Address:  "https://localhost:4321/api/v2/config",
```

Comment:
This seems to hardcode the Address. I think we need to provide the configured address for the console service here.

Reply:
Updated so we return the console URL e.g.

```json
{
    "readOnly": true,
    "configs": [
        {
            "configType": "auto",
            "name": "Topaz Config",
            "address": "https://localhost:8080/api/v2/config",
            "authorizerServiceUrl": "https://localhost:8383",
            "authorizerApiKey": "",
            "directoryServiceUrl": "https://localhost:9393",
            "directoryApiKey": "",
            "directoryTenantId": "",
            "directoryReaderServiceUrl": "https://localhost:9393",
            "directoryWriterServiceUrl": "https://localhost:9393",
            "directoryImporterServiceUrl": "https://localhost:9393",
            "directoryExporterServiceUrl": "https://localhost:9393",
            "directoryModelServiceUrl": "https://localhost:9393"
        }
    ]
}
```